src/shared/dissect-image.c

```diff
@@ -1063,6 +1063,24 @@ static int dissect_image(
                         log_debug("Dissecting %s partition with label %s and UUID %s.",
                                   strna(partition_designator_to_string(type.designator)), strna(label), SD_ID128_TO_UUID_STRING(id));
 
+                        if (FLAGS_SET(pflags, SD_GPT_FLAG_NO_AUTO) && type.designator != PARTITION_ESP) {
```

Comment:
could you add a comment that underlines that the NO_AUTO flag is not defined for ESPs, as ESP is mandated by UEFI spec, not boot laoder spec